### PR TITLE
Implement selectRegionLayer and variablized layers 

### PR DESCRIPTION
**Pull request by EricHanLiu**
Opening this pull request for further discussion to be based here regarding how the iterative classification will be done with the CNN

[Accompanying issue](https://github.com/DDMAL/pixel_wrapper/issues/22)

### CODE REVIEW

File: source/js/plugins/pixel-wrapper.js [hunk 1]
```diff
@@ -145,9 +147,15 @@ export class PixelWrapper
...
             height = this.pixelInstance.core.publicInstance.getPageDimensionsAtZoomLevel(pageIndex, maxZoom).height;
 
-        // Highlight whole image for background layer
-        let rect = new Rectangle(new Point(0, 0, pageIndex), width, height, "add");
-        backgroundLayer.addShapeToLayer(rect);
+        // Add select regions to backgroundLayer
+        selectRegionLayer.shapes.forEach((shape) => {
+            let x = shape.origin.getCoordsInPage(maxZoom).x,
+                y = shape.origin.getCoordsInPage(maxZoom).y,
+                rectWidth = shape.relativeRectWidth * 2^maxZoom,
```
Comment (vigliensoni) [inline source/js/plugins/pixel-wrapper.js]: Be consistent with the spacing of math operators

Comment (EricHanLiu) [reply source/js/plugins/pixel-wrapper.js]: Will do!

Comment (EricHanLiu) [reply source/js/plugins/pixel-wrapper.js]: Wait, this has been changed to `Math.pow(2, maxZoom)` already. Not sure why it's showing like so here

File: source/js/plugins/pixel-wrapper.js [hunk 2]
```diff
@@ -150,30 +208,31 @@ export class PixelWrapper
...
             while (cnt--) { 
                 if (row >= height)
                     break;
                 if (col < width) {
-                    let data = pixelCtx.getImageData(col, row, 1, 1).data,
-                        colour = new Colour(data[0], data[1], data[2], data[3]);
-                    if (colour.alpha !== 0) { 
-                        let currentPixel = new Rectangle(new Point(col, row, pageIndex), 1, 1, "subtract");
+                    let data = pixelCtx.getImageData(col, row, 1, 1).data;
+                    // data[3] is alpha
```
Comment (vigliensoni) [inline source/js/plugins/pixel-wrapper.js]: Does the ```data``` array follow RGB+alpha? Please remind that in the code.

Comment (EricHanLiu) [reply source/js/plugins/pixel-wrapper.js]: Done!

**Pull request merged**